Updating the included webserver to support searching

traetox opened the pull request:
This includes an updated golang webserver that holds an in-memory index and search API so that the documentation can perform some basic searching and link to anchor tags.

The in-memory index is NOT efficient, but it works and the data set is small.

djfritz commented on serve.go:
```diff
@@ -1,10 +1,112 @@
+
+func main() {
+	flag.Parse()
+	if *lp == 0 || *lp > 0xffff {
```
This is somewhat unnecessary  as ListenAndServe() will fail when it attempts to bind.

traetox replied:
i know, but i like throwing a super specific error.

traetox commented:
Updated to lowercase everything so queries don't have to worry about case.

We also updated the index structure to have some more metadata that will make building a search interface better.

traetox commented:
this: `curl -s --header "Content-Type: application/json"   --request POST   --data '{"value": "enumerated value JSON strict"}' http://localhost:3001/api/search`

now returns this:
```
{
  "links": [
    {
      "Page": "Search Extraction Modules",
      "Heading": "Universal Flags",
      "Link": "#!search/extractionmodules.md#Universal_Flags"
    },
    {
      "Page": "Search Processing Modules",
      "Heading": "Universal Flags",
      "Link": "#!search/processingmodules.md#Universal_Flags"
    },
    {
      "Page": "JSON",
      "Heading": "Empty Fields and the Strict Flag",
      "Link": "#!search/json/json.md#Empty_Fields_and_the_Strict_Flag"
    },
    {
      "Page": "JSON",
      "Heading": "Supported Options",
      "Link": "#!search/json/json.md#Supported_Options"
    }
  ]
}
```

djfritz commented on search/dump/dump.md:
```diff
@@ -0,0 +1,70 @@
+## Dump
+
+The `dump` module is designed to treat resources as query-able data, meaning that you can inject a CSV or lookup table into the query pipeline and perform operations on the rows as if they are full fledged search entries.  Dump can be particularly useful when building dashboards and views that need to operate on data that has already been imported from some other static source or aggregated used  ...
```
s/query-able/queryable/

djfritz commented on search/dump/dump.md:
```diff
@@ -0,0 +1,70 @@
+
+### Filtering and Column Inclusion
+
+The `dump` module will extract all columns from a CSV or lookup table resource if no columns are specified or if all specified columns have filters operations.  For example `dump -r hosts hostname` means only extract the hostname column, where `dump -r hosts hostname == "ad.example.com"` means extract all columns because every column specified has a filter operation attached.  The query `dump -r  ...
```
s/For example/For example,/

djfritz commented on search/dump/dump.md:
```diff
@@ -0,0 +1,70 @@
+
+### Example Queries
+
+Dumps an entire resource into the pipeline and populates the table module:
```
s/Dumps/Dump/
